managers/models.py

```diff
@@ -59,6 +67,64 @@ def clean(self) -> None:
             raise ValidationError(message=ProjectConstants.STOP_DATE_VALIDATION_ERROR_MESSAGE.value)
 
 
+class ExpectedHoursManager(models.Manager):
+    def get_expected_hours_sum_for_date(self, for_date: date, excluded_id: Optional[int] = None) -> timedelta:
+        query_for_date = self.filter(date=for_date)
+
+        if excluded_id is not None:
+            query_for_date = query_for_date.exclude(pk=excluded_id)
+
+        return query_for_date.aggregate(expected_hours_sum=Coalesce(models.Sum("value"), timedelta()))[
+            "expected_hours_sum"
+        ]
+
+
+class ExpectedHours(BaseModel):
+    date = models.DateField(help_text=ExpectedHoursConstants.MESSAGE_FOR_CORRECT_DATE_FORMAT.value)
+    value = models.DurationField(
+        validators=[
+            MinValueValidator(ExpectedHoursConstants.MIN_EXPECTED_HOURS.value),
+            MaxValueValidator(ExpectedHoursConstants.MAX_EXPECTED_HOURS.value),
+        ]
+    )
+    user = models.ForeignKey(CustomUser, on_delete=models.CASCADE, related_name="expected_hours")
+    project = models.ForeignKey(Project, on_delete=models.CASCADE, related_name="expected_hours")
+
+    objects = ExpectedHoursManager()
+
+    def check_if_date_only_when_project_is_active(self) -> None:
+        if self.date < self.project.start_date:
+            raise ValidationError(message=ExpectedHoursStrings.EXPECTED_HOURS_DATE_BEFORE_PROJECT_START.value)
+        if self.project.stop_date is not None:
+            if self.date > self.project.stop_date:
+                raise ValidationError(message=ExpectedHoursStrings.EXPECTED_HOURS_DATE_AFTER_PROJECT_FINISHED.value)
+        if self.project.suspended:
+            raise ValidationError(message=ExpectedHoursStrings.EXPECTED_HOURS_FOR_PROJECT_SUSPENDED.value)
+
+    def check_if_expected_hours_for_user_less_than_24h_per_day(self) -> None:
+        if (
+            self.user.expected_hours.get_expected_hours_sum_for_date(self.date, self.pk) + self.value
+            > ExpectedHoursConstants.MAX_EXPECTED_HOURS.value
+        ):
+            raise ValidationError(message=ExpectedHoursStrings.EXPECTED_HOURS_EXCEED_24H_PER_DAY_PER_USER.value)
+
+    def check_if_user_works_in_project(self) -> None:
+        if not (
+            self.user.manager_projects.filter(pk=self.project.pk).exists()
+            or self.user.projects.filter(pk=self.project.pk).exists()
+        ):
+            raise ValidationError(message=ExpectedHoursStrings.USER_NOT_IN_THE_PROJECT.value)
```

Comment:
All three methods should be private. Second thing is that you need to always make proper order of methods. You need to look at file like you would read for example a book. I'll try to illustrate you example:
```python
def funA():
    funB()
    funC()
    funD()

def funB():
    funE()

def funE():
    print("funE")

def funC():
    print("funC")

def funD():
    print("funD")
```

The same thing you need to do with methods in this file. First need to be `clean()` method and under it all methods which are called inside `clean`

Reply:
Ok. I will remember the order of methods in the future :)